src/runtime-rs/crates/runtimes/src/static_resource.rs

```diff
@@ -0,0 +1,167 @@
+                    quota: Some(quota),
+                    ..Default::default()
+                };
+                // although it may not be actually a linux container, we are only using the calculation inside
+                // LinuxContainerCpuResources::try_from to generate our vcpu number
+                if let Ok(cpu_resource) = LinuxContainerCpuResources::try_from(&cpu) {
+                    vcpu = get_nr_vcpu(&cpu_resource);
+                }
+                mem_mb = convert_memory_to_mb(memory);
+            }
+            // single container, from container spec
+            _ => {
+                if let Some(linux) = &spec.linux {
+                    if let Some(resource) = &linux.resources {
+                        if let Some(cpu) = &resource.cpu {
+                            if let Ok(cpu_resource) = LinuxContainerCpuResources::try_from(cpu) {
+                                vcpu = get_nr_vcpu(&cpu_resource);
+                            }
+                        }
+                        if let Some(mem) = &resource.memory {
+                            let memory = mem.limit.unwrap_or(0);
+                            mem_mb = convert_memory_to_mb(memory);
+                        }
+                    }
+                }
+            }
+        }
+        info!(
+            sl!(),
+            "static resource mgmt result: vcpu={}, mem_mb={}", vcpu, mem_mb
+        );
+        Ok(Self { vcpu, mem_mb })
+    }
+}
+
+// StaticResourceManager is responsible for static resource management
+//
+// static resource management sizing information is optionally provided, either by
+// upper layer runtime (containerd / crio) or by the container spec itself (when it
+// is a standalone single container such as the one started with *docker run*)
+//
+// the sizing information uses three values, cpu quota, cpu period and memory limit,
+// and with above values it calculates the # vcpus and memory for the workload and
+// add them to default value of the config
+#[derive(Clone, Copy, Debug)]
+pub struct StaticResourceManager {
+    resource: StaticResource,
+}
+
+impl StaticResourceManager {
+    pub fn new(spec: &oci::Spec) -> Result<Self> {
+        Ok(Self {
+            resource: StaticResource::try_from(spec)
+                .context("failed to construct static resource")?,
+        })
+    }
+
+    pub fn setup_config(&self, config: &mut TomlConfig) -> Result<()> {
+        // update this data to the hypervisor config for later use by hypervisor
+        let hypervisor_name = &config.runtime.hypervisor_name;
+        let mut hv = config
+            .hypervisor
+            .get_mut(hypervisor_name)
+            .context("failed to get hypervisor config")?;
+        hv.cpu_info.default_vcpus += self.resource.vcpu as i32;
+        hv.memory_info.default_memory += self.resource.mem_mb;
+        Ok(())
+    }
+}
+
+fn get_nr_vcpu(resource: &LinuxContainerCpuResources) -> u32 {
+    if let Some(v) = resource.get_vcpus() {
```

Comment:
Use `map_or` is simple and easy to understand:

`resource.get_vcpus().map_or(0, |v| v as u32)`

Reply:
Fixed